src/components/UserData/DropdownList.tsx

```diff
@@ -0,0 +1,41 @@
+import classNames from 'classnames';
+import React from 'react';
+import { User } from '../../types/User';
+import { Loader } from '../Loader';
+
+type Props<T> = {
+  list: T[] | null
+  chosenElem: T | null
+  handler: (elem: (T | null)) => () => void
```

Comment:
The function passed to the handler prop is immediately invoked which can lead to performance issues and unintended behaviors. It should only be invoked when the onClick event is triggered. Instead of passing a function that is immediately invoked, pass a function that will return another function when it's executed.